frame/contracts/src/lib.rs

```diff
@@ -1216,11 +1216,11 @@ impl<T: Config> Pallet<T> {
 		<T::Currency as Inspect<AccountIdOf<T>>>::minimum_balance()
 	}
 
-	/// Convert a 1D Weight to a 2D weight.
+	/// Convert gas_limit from 1D Weight to a 2D Weight.
 	///
-	/// Used by backwards compatible extrinsics. We cannot just set the proof to zero
-	/// or an old `Call` will just fail.
-	fn compat_weight(gas_limit: OldWeight) -> Weight {
+	/// Used by backwards compatible extrinsics. We cannot just set the proof_size weight limit to
+	/// zero or an old `Call` will just fail with OutOfGas.
+	fn compat_weight_limit(gas_limit: OldWeight) -> Weight {
 		Weight::from(gas_limit).set_proof_size(u64::from(T::MaxCodeLen::get()) * 2)
```

Comment:
I know not changes in this PR but since we are at it.

```suggestion
		Weight::from_parts(gas_limit, u64::from(T::MaxCodeLen::get()) * 2)
```